pkg/releaselock/controller.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rpc
+
+import (
+	"context"
+	"os"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apiError "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/gcp-filestore-csi-driver/pkg/util"
+
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+)
+
+const (
+	gceInstanceIDKey = "container.googleapis.com/instance_id"
+	leaseName        = "filestore-csi-storage-gke-io-node"
+	// Root CA configmap in each namespace.
+	rootCA = "kube-root-ca.crt"
+
+	defaultLeaderElection = true
+	defaultLeaseDuration  = 15 * time.Second
+	defaultRenewDeadline  = 10 * time.Second
+	defaultRetryPeriod    = 2 * time.Second
+	defaultSyncPeriod     = 3 * time.Minute
```

Comment:
lets start with a 1 Minute default and check the behavior in a ~100 node cluster

Reply:
done